sources/CMakeLists.txt

```
@@ -12,6 +12,8 @@ add_library(neko-top STATIC)
 # Add the local neko extension library
 add_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/neko_ext)
 
+# Add the mapping extension library
+add_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/mapping_functions)
```

was this the right move?
ie, should we be doing this with everything? `adjoint` , `problem` etc

or are we only doing this because it contains a backend?

---

I did that because `neko_ext` is meant to be for things that should be moved across to neko in time. 
You can add it that way if you like. It might be good to do since the src cmakelists is already getting really big at this point 😅

---

Please add an empty line bellow to follow the structure of the files.
```suggestion
add_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/mapping_functions)

```